diff --git a/lib/routes/huxiu/util.ts b/lib/routes/huxiu/util.ts
--- a/lib/routes/huxiu/util.ts
+++ b/lib/routes/huxiu/util.ts
@@ -178,7 +178,7 @@ const fetchItem = async (item) => {
     const { data: detailResponse } = await got(item.link);
 
     const state = parseInitialState(detailResponse);
-    const data = state.briefStoreModule?.brief_detail.brief ?? state.articleDetail?.articleDetail ?? undefined;
+    const data = state?.briefStoreModule?.brief_detail?.brief ?? state?.articleDetail?.articleDetail ?? undefined;
 
     if (!data) {
         return item;
@@ -33,10 +33,11 @@ const cleanUpHTML = (data) => {
     $('em.vote__bar, div.vote__btn, div.vote__time').remove();
     $('p img').each((_, e) => {
         e = $(e);
+        const imageSrc = e.prop('src') ?? e.prop('_src');
         e.parent().replaceWith(
             art(path.join(__dirname, 'templates/description.art'), {
                 image: {
-                    src: (e.prop('src') ?? e.prop('_src')).split(/\?/)[0],
+                    src: imageSrc ? imageSrc.split(/\?/)[0] : undefined,
                     width: e.prop('data-w'),
                     height: e.prop('data-h'),
                 },
@@ -271,6 +272,13 @@ const processAudioInfo = (info) => {
     const linkKey = `audio_${quality}path`;
     const sizeKey = `audio_${quality}size`;
 
+    if (!info[linkKey]) {
+        return {
+            processed: undefined,
+            processedItem: {},
+        };
+    }
+
     const processed = {
         duration: info.format_length_new ?? info.format_length,
         size: Object.hasOwn(info, sizeKey) ? info[sizeKey] : undefined,
@@ -428,6 +436,13 @@ const processVideoInfo = (info) => {
     const linkKey = `${quality}_link`;
     const sizeKey = `origin_${quality}_size`;
 
+    if (!info[linkKey]) {
+        return {
+            processed: undefined,
+            processedItem: {},
+        };
+    }
+
     const processed = {
         duration: info.duration ?? info.origin_duration,
         poster: info.cover ?? info.custom_cover_path ?? info.gif_path,
